### PR TITLE
TBM-5338 - added bare bones auto complete input

### PR DESCRIPTION
**Pull request by troyblank**

### CODE REVIEW

File: assets/js/mockData/states.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    'New Hampshire',
+    'New Jersey',
+    'New Mexico',
+    'New York',
+    'North Carolina',
+    'North Dakota',
+    'Ohio',
+    'Oklahoma',
+    'Oregon',
+    'Pennsylvania Rhode Island',
```
Comment (jhweaver) [inline assets/js/mockData/states.js]: Nope.

Comment (troyblank) [inline assets/js/mockData/states.js]: daaamn! thanks lol

Comment (troyblank) [inline assets/js/mockData/states.js]: @jhweaver fixed!

File: components/form/autoComplete.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    onEnterKey() {
+        this.selection = this.state.options[this.state.focus] ? this.state.options[this.state.focus] : '';
+    }
+
+    updateOptions(value) {
+        let regex;
+
+        this.state.options = [];
+        this.props.options.forEach((option) => {
+            regex = new RegExp(value, 'gi');
```
Comment (jhweaver) [inline components/form/autoComplete.js]: Are we sure we want to do these with regexs? We won't be able to search for some special characters. I'm not sure what kind of characters are normally in there, so I'm not sure if it's really a concern.

Comment (troyblank) [inline components/form/autoComplete.js]: Yea not fully sure currently; this iteration is very basic so I am fine with regexes for now.

**Pull request merged**